app/constantData/demoData.ts

```diff
@@ -0,0 +1,17 @@
+export const courseDetails = {
+  id: 1,
+  teacherName: "Mr. John Doe",
+  Section: "A",
+  value: 235,
+  bookList: ["Book 01", "Book 02", "Book 03"],
+  authorList: ["Author 01", "Author 02", "Author 03"],
+};
+
+function generateCourseList(totalItems: number) {
```

Comment:
Make generateCourseList a pure function

Ensure it doesn’t rely on external variables (courseDetails) by passing it as a parameter.
```suggestion
const generateCourseList = (totalItems: number, courseDetails: object) =>
```

Reply:
certainly. changed